### PR TITLE
Components: Update accessibility of Checklist

### PR DESCRIPTION
**Pull request by ryelle**
Over in Store, we're about to use this component for our setup checklist (#23067), but there are a few accessibility issues. This PR fixes most of them - the main updates are:

- Increase contrast of all the light-grey text by using the `$gray-text-min` color
- Switch the heading levels to `h2`/`h3`, to match the site checklist page (which has just an `h1`)
- Use the Focusable component on the toggle icons so that they can be visually keyboard-accessible
- Switch the task titles to use Buttons so they can be appropriately keyboard + screen reader accessible
- This PR also fixes an issue for #23067, where we need a prop for custom button text per-task (rather than Do it! for all)

I did _not_ adjust the contrast of the blue task title text, even though that combination is not sufficiently contrasted.

| Before | After |
| --- | --- |
| current | updated |

If this is merged, it would be great to use the new `buttonText` prop for custom button text on the Site Checklist page. As it's possible to navigate a page just by listing form controls, screen reader users will just hear a bunch of buttons that say "Do it!". The addition of the button on the task title helps this, but now there are 2 buttons for the same action, which is not ideal. If we can move over to using custom `buttonText`, maybe we can remove the button from the task title in the future.

**To test**

- Check out the checklist on `/devdocs/design/checklist` or `/checklist/:site` (for a wp.com simple site)
- Tab through the checklist items, make sure you can get to all of them
- [Try out VoiceOver + Safari](https://webaim.org/articles/voiceover/) & make sure you can get through the checklist items

### CODE REVIEW

**Comment by ryelle** [comment]
> It doesn't do anything if I hit enter on the checkmark.

That's because the toggle action doesn't do anything on completed checkmarks - you get the same result from clicking on them. Enter works correctly on the uncompleted toggles, and on the completed items in [the devdocs example](https://calypso.live/devdocs/design/checklist?branch=update/a11y-checklist).

**Comment by fditrapani** [comment]
> That's because the toggle action doesn't do anything on completed checkmarks

That's fair and should probably be worked on in another PR.

File: client/components/checklist/checklist-task.jsx [hunk 1]
```diff
@@ -16,12 +16,14 @@ import Gridicon from 'gridicons';
  */
 import Button from 'components/button';
 import Card from 'components/card';
+import Focusable from 'components/focusable';
```
Comment (timmyc) [inline client/components/checklist/checklist-task.jsx]: Wait... i could have sworn someone told me to never use this component 😃

File: client/components/checklist/checklist-task.jsx [hunk 2]
```diff
@@ -53,6 +55,7 @@ export class ChecklistTask extends PureComponent {
 			title,
 			translate,
 		} = this.props;
+		const { buttonText = translate( 'Do it!' ) } = this.props;
```
Comment (timmyc) [inline client/components/checklist/checklist-task.jsx]: awesome, thank you.

**Pull request merged**